integration/experiment/power_sweep/gen_policy_recommendation.py

```diff
@@ -0,0 +1,287 @@
+                if not firstTerm:
+                    if cc < 0:
+                        expression += " - "
+                        cc *= -1
+                    else:
+                        expression += " + "
+                expression += str(cc) + tt
+            # TODO classname
+            return "<%s %s>" % ("Polyfitmodel", expression)
+        else:
+            return "<%s (untrained)>" % "Polyfitmodel"  #TODO classname
+
+
+class CubicPowerModel(PolynomialFitPowerModel):
+    def __init__(self):
+        super().__init__(degree=3)
+
+
+def extract_from_reports(path, region_filter):
+    simplified_data = []
+    
+    try:
+        df = geoio.RawReportCollection("*report", dir_name=path).get_df()
+    except RuntimeError:
+        sys.stderr.write("<geopm> Error: No report data found in " + path + \
+                "; run a power sweep before using this analysis.\n")
+        sys.exit(1)
+    for profile in df['Profile'].unique():
+        # profile will be something like "unifiedmodel_power_governor_250_1"
+        power_limit = int(profile.split('_')[-2])  # TODO - access the policy
+        df_profile = df[df['Profile'] == profile]
+        for host in df_profile['host'].unique():
+            df_profile_host = df_profile[df_profile['host'] == host]
+            total_rt = 0
+            total_en = 0
+            for region_name in df_profile_host['region'].unique():
+                if region_filter and region_name not in region_filter:
+                    continue
+                region = df_profile_host[df_profile_host['region'] == region_name]
+                runtime = sum(region['runtime (sec)'])
+                energy = sum(region['package-energy (joules)'])
+    
+                total_rt += runtime
+                total_en += energy
+    
+            simplified_data.append((power_limit, total_rt, total_en))
+    return simplified_data
+
+
+def dump_simplified_data(simplified_data, fname):
+    raw_out = open(fname, "w")
+    for pl, rt, en in sorted(simplified_data):
+        raw_out.write("%d %f %f\n" % (pl, rt, en))
+    raw_out.close()
+
+
+def dump_stats_summary(simplified_data, fname):
+    count = {}
+    for pl, rt, en in sorted(simplified_data):
+        if pl in count:
+            count[pl] += 1
+        else:
+            count[pl] = 1
+
+    rtx = {} ; rtxx = {}
+    enx = {} ; enxx = {}
+    for pl in count:
+        rtx[pl] = 0 ; rtxx[pl] = 0
+        enx[pl] = 0 ; enxx[pl] = 0
+
+
+    for pl, rt, en in sorted(simplified_data):
+        rtx[pl] += rt
+        rtxx[pl] += rt**2
+        enx[pl] += en
+        enxx[pl] += en**2
+
+    for pl in count:
+        rtx[pl] /= count[pl]
+        rtxx[pl] /= count[pl]
+        enx[pl] /= count[pl]
+        enxx[pl] /= count[pl]
+
+    stats_out = open(fname, "w")
+    for pl in sorted(count.keys()):
+        stats_out.write("%d %f %f %f %f\n" % (pl, rtx[pl], (rtxx[pl] - rtx[pl]**2)**.5, enx[pl], (enxx[pl] - enx[pl]**2)**.5))
```

Comment:
Please refactor all of your ```write()``` calls (and any other string manipulation that uses the % syntax) to use ```"".format()```.  That's the style we use for all of our string manipulation.

Reply:
Done.